src/core/tests/Test_ConfigurePatchingProcessor.py

```diff
@@ -170,6 +172,91 @@ def test_operation_fail_for_configure_patching_request_for_apt(self):
         self.assertTrue(substatus_file_data[0]["status"] == Constants.STATUS_ERROR.lower())
         runtime.stop()
 
+    def test_configure_patching_with_assessment_mode_by_platform(self):
+
+        # create and adjust arguments
+        argument_composer = ArgumentComposer()
+        argument_composer.operation = Constants.CONFIGURE_PATCHING
+        argument_composer.patch_mode = Constants.PatchModes.IMAGE_DEFAULT
+        argument_composer.assessment_mode = Constants.AssessmentModes.AUTOMATIC_BY_PLATFORM
+
+        # create and patch runtime
+        runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.APT)
+        runtime.package_manager.get_current_auto_os_patch_state = runtime.backup_get_current_auto_os_patch_state
+        runtime.package_manager.os_patch_configuration_settings_file_path = os.path.join(runtime.execution_config.config_folder, "20auto-upgrades")
+        runtime.set_legacy_test_type('HappyPath')
+
+        # mock os patch configuration
+        os_patch_configuration_settings = 'APT::Periodic::Update-Package-Lists "1";\nAPT::Periodic::Unattended-Upgrade "1";\n'
+        runtime.write_to_file(runtime.package_manager.os_patch_configuration_settings_file_path, os_patch_configuration_settings)
+
+        # execute Core
+        CoreMain(argument_composer.get_composed_arguments())
+
+        # check telemetry events
+        self.__check_telemetry_events(runtime)
+
+        # check status file for configure patching patch mode
+        with runtime.env_layer.file_system.open(runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)[0]["status"]["substatus"]
+
+        # check status file for configure patching patch state
+        self.assertEquals(len(substatus_file_data), 1)
+        self.assertTrue(substatus_file_data[0]["name"] == Constants.CONFIGURE_PATCHING_SUMMARY)
+        self.assertTrue(substatus_file_data[0]["status"] == Constants.STATUS_SUCCESS.lower())
```

Comment:
nit: `.lower()` should be on the left hand side since it is a constant on the right

Reply:
Do you mean it should be this way: 

self.assertTrue(Constants.STATUS_SUCCESS.lower() == substatus_file_data[0]["status"]) ? or something else?

If you mean something like this lower(Constants.STATUS_SUCCESS), that's is not a function in python.

Reply:
I was thinking something like this:

```py
self.assertTrue(substatus_file_data[0]["status"].lower() == Constants.STATUS_SUCCESS)
```

Reply:
That wouldn't work because Constants.STATUS_SUCCESS is set to 'Success' and the one on left is already 'success'. 

With this: self.assertTrue(substatus_file_data[0]["status"].lower() == Constants.STATUS_SUCCESS)
the equation would be 'success' == 'Success', which will fail

But you've raised a good point. For equality we need to ensure both values are exactly same in terms of casing too. So we should have lower() on both sides just in case, to avoid any case mismatch

Reply:
Oh, I see, yep that makes sense. Having it on both sides sounds like a good idea to avoid any other case mismatches.